src/Dialect/ONNX/Rewrite.cpp

```diff
@@ -874,6 +874,78 @@ class PowToMulRewritePattern : public OpRewritePattern<ONNXPowOp> {
   int64_t maxPower;
 };
 
+// Rewrite a pattern like the following:
+//
+// %shape = onnx.Concat(%dim1, %dim2)
```

Comment:
A question that I believe I know the answer, but just to double check.

The pattern that is optimized here is really `expand -> unsqueezed`. Because our current approach is to always have a concat of the dimensions (that includes one or more runtime dims), then we will always find a `concat` of the shape, correct?
And if the shape itself is a constant, then constant propagation will anyway get rid of all 3 and generate one big constant for `%u`, so the constant shape is not relevant to this pattern.

Thanks for double checking.

Reply:
Yes, you are right, but we still double-check if the shape is really from concat (it's maybe the case when SimplifiedShapeRelatedOp has not yet called and we still have ShapeOp instead of Concat. Perhaps, in the future we will always decompose ShapeOp into Concat), so that we can get each dim and insert unsqueezing dims.